docs/fml-spec.md

```diff
@@ -0,0 +1,576 @@
+      item-thumbnail:
+        description: "The icon that appears in the Spotlight search results.
+          Note that changing this does not change already indexed content."
+        type: ThumbnailType
+        default: letter
+    defaults:
+      - value: { item-thumbnail: screenshot, max-age-in-days: 56.0 }
+```
+
+The `defaults` list of `value`s set the default value of `item-thumbnail` to `screenshot`, and `max-age-in-days` to `56.0`.
+
+Adding new feature blocks to the `defaults` list patches the existing defaults further:
+
+```yaml
+    defaults:
+      - value: { item-thumbnail: screenshot, max-age-in-days: 56.0 }
+      - value: { max-age-in-days: 64.0 }
+```
+
+In this case, the default value for `item-thumbnail` is now `screenshot`, and `max-age-in-days` is `64.0`.
+
+### Feature defaults and channels
+
+For release engineering reasons, or marketing reasons, we might want to ensure that the feature isn't enabled by default for the release population of users.
+
+However, for quality reasons, we might want a different configuration for our nightly audience.
+
+This is achieved by using channels.
+
+```yaml
+channels:
+  - nightly
+  - beta
+  - release
+features:
+  spotlight-search:
+    variables:
+      enabled:
+        description: "…"
+        type: Bool
+        default: false
+    defaults:
+      - channel: nightly
+        value: { enabled: true }
+```
+
+The `channel` property specifies which build flavour the default applies. In this example, the `nightly` version of the app set the `enabled` variable to `true`; all other versions of the app have `enabled` as `false`.
+
+In this way, app developers can use the feature manifest as a replacement for other more adhoc feature flag solutions.
+
+### Feature defaults and runtime targeting
+
+> ⚠️ Unimplemented. This is intended to use the same JEXL targeting used elsewhere in Nimbus SDK.
+
+Occassionally, we would like to vary a configuration based upon the device. In this example, the `homepage` feature has sections for both Pocket and Topsites.
+
+```yaml
+features:
+  homepage:
+    variables:
+      pocket-enabled:
+        description: If true, show stories from Pocket. This is only available in certain territories.
+        type: Bool
+        default: false
+      topsites-enabled:
+        description: If true, show tiles from the users most recent and frequently visited pages.
+        type: Bool
+        default: false
+    defaults:
+      - channel: nightly
+        value: { topsites-enabled: true }
+      - channel: nightly
+        targeting: locale in ['en-US', 'de-DE', 'en-GB']
+        value: { pocket-enabled: true }
+```
+
+In this example, `pocket-enabled` defaults to `true` only on the `nightly` channel and when the device locale is in `US`, `DE` or `GB`.
+
+## Additional types
+
+### Primitive types
+
+Primitive types supported:
+
+ * `String`
+ * `Boolean` or `Bool`. Both Kotlin and Swift's spelling is supported.
+ * `Int`
+
+> ⚠️ Unimplemented
```

Comment:
nit: Should this be under "Double", just to be consistent with where the other "Unimplemented" were placed